### PR TITLE
FIX-#5819: Fix np.argmax/argmin on 1D arrays

### LINKED ISSUE #5819: BUG: np.argmax/argmin on 1D arrays are broken
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
np.argmax(np.array([-87.434,-90.908, -87.152, -84.903]))
# Returns 0 (should return 3)
```

### Issue Description

A few internal logic bugs in argmax/argmin are producing the incorrect result.

### Expected Behavior

Return the same value as numpy does.

### Error Logs

<details>

```python-traceback

Replace this line with the error backtrace (if applicable).

```

</details>

### Installed Versions

<details>
INSTALLED VERSIONS
------------------
commit           : d5f3f723bbe0f941b28e8a09ef1fac3e4579f7a9
python           : 3.10.4.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 22.2.0
Version          : Darwin Kernel Version 22.2.0: Fri Nov 11 02:03:51 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T6000
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

Modin dependencies
------------------
modin            : 0.19.0+11.gd5f3f723.dirty
ray              : 1.13.0
dask             : 2022.8.0
distributed      : 2022.8.0
hdk              : None

pandas dependencies
-------------------
pandas           : 1.5.3
numpy            : 1.23.1
pytz             : 2022.1
dateutil         : 2.8.2
setuptools       : 61.2.0
pip              : 22.1.2
Cython           : None
pytest           : 7.1.2
hypothesis       : None
sphinx           : 5.1.1
blosc            : None
feather          : 0.4.1
xlsxwriter       : None
lxml.etree       : 4.9.1
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.4.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : None
fastparquet      : 0.8.2
fsspec           : 2022.7.1
gcsfs            : None
matplotlib       : 3.5.3
numba            : None
numexpr          : 2.8.3
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : 0.17.8
pyarrow          : 8.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.7.1
scipy            : 1.9.0
snappy           : None
sqlalchemy       : 1.4.40
tables           : 3.7.0
tabulate         : None
xarray           : 2022.6.0
xlrd             : 2.0.1
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by noloerino**
## What do these changes do?

Fixes a bug where argmax/argmin on 1D arrays were giving incorrect results. There may still be bugs left over for 2D arrays, which we plan to address later with more comprehensive tests for our numpy API.

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5819  
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Review by RehanSD — DISMISSED** [review]
LGTM!

File: modin/numpy/test/test_array_math.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    ids=["ints", "floats", "floats with nan"],
+)
+def test_argmax_argmin(data):
+    numpy_result = numpy.argmax(numpy.array(data))
+    modin_result = np.argmax(np.array(data))
+    numpy.testing.assert_array_equal(modin_result, numpy_result)
+
+    numpy_result = numpy.argmin(numpy.array(data))
+    modin_result = np.argmin(np.array(data))
+    numpy.testing.assert_array_equal(modin_result, numpy_result)
```
Comment (anmyachev) [inline modin/numpy/test/test_array_math.py:36]:
```suggestion
@pytest.mark.parametrize("op", ["argmin", "argmax"])
def test_argmax_argmin(data, op):
    numpy_result = getattr(numpy, op)(numpy.array(data))
    modin_result = getattr(np, op)(np.array(data))
    numpy.testing.assert_array_equal(modin_result, numpy_result)
```

File: modin/numpy/arr.py [hunk 2]
```diff
@@ -2179,9 +2179,9 @@ def argmax(self, axis=None, out=None, keepdims=None):
                 raise numpy.AxisError(1, 1)
             if self._query_compiler.isna().any(axis=1).any(axis=0).to_numpy()[0, 0]:
                 na_row_map = self._query_compiler.isna().any(axis=1)
-                result = na_row_map.idxmax().to_numpy()[0, 0]
+                result = na_row_map.idxmax()
             else:
-                result = self._query_compiler.idxmax(axis=1)
+                result = self._query_compiler.idxmax(axis=0)
```
Comment (anmyachev) [inline modin/numpy/arr.py:2184]: Why was the axis changed?

Comment (noloerino) [reply modin/numpy/arr.py:2184]: The opposite axis was incorrect. Here, query_compiler is an Nx1 array, and we need axis=0 to reduce it to 1x1. This was apparently already fixed for argmin on [line 2255](https://github.com/modin-project/modin/pull/5820/files#diff-a5565cfb004b107c88dbae43a112a1821d007fdb560eef82c66c25d57c73cb4fR2255).

Comment (anmyachev) [reply modin/numpy/arr.py:2184]: I see, thanks!

**Review by anmyachev — APPROVED** [review]

**Pull request merged**